types/react-simple-oauth2-login/index.d.ts

```diff
@@ -0,0 +1,34 @@
+// Type definitions for react-simple-oauth2-login 0.5
+// Project: https://github.com/bhubr/react-simple-oauth2-login#readme
+// Definitions by: Kostas Antonopoulos <https://github.com/kostassite>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as React from 'react';
+
+export interface OAuth2LoginProps {
+    id?: string;
+    authorizationUrl: string;
+    clientId: string;
+    redirectUri: string;
+    responseType: 'code' | 'token';
+    onSuccess: (data: Record<string, any>) => void;
+    onFailure: (err: Error) => void;
+    buttonText?: string;
+    children?: React.ReactChildren;
+    popupWidth?: number;
+    popupHeight?: number;
+    className?: string;
+    render?: (props: {
+        className: string,
+        buttonText: string,
+        children: React.ReactChildren,
+        onClick: () => void
+    }) => void;
+    isCrossOrigin?: boolean;
+    onRequest?: () => void;
+    scope?: string;
+    state?: string;
+    extraParams?: Record<string, any>;
+}
+
+export default function OAuth2Login(props: OAuth2LoginProps): JSX.Element;
```

Comment:
The upstream package is CJS, so we shouldn't use `export default` here. Additionally, the upstream repo is a class.

I think you should write this like:

```ts
export = OAuth2Login;

declare class OAuth2Login extends React.Component<OAuth2Login.OAuth2LoginProps> {}

declare namespace OAuth2Login {
    interface OAuth2LoginProps {
        // ...
    }
}
```

If you want an example, you check out `react-popover`.

Reply:
Apologies, I misread the package; it really does export `default`.